src/pages/signUp/signUpSteps/Welcome.tsx

```diff
@@ -0,0 +1,95 @@
+import IconButton from "@mui/material/IconButton";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";
+import { Box, Stack, useTheme } from "@mui/material";
+import { FilledButton, Typography } from "elements";
+import { FunctionComponent, useEffect, useState } from "react";
+import { useAuthenticatedRedirect } from "common";
+import { FormikValues, useFormikContext } from "formik";
+import {
+  FacebookLogin,
+  GoogleLogin,
+  LinkedInLogin,
+  TextInputField,
+} from "components";
+
+
+const SignUp: FunctionComponent = () => {
+  const theme = useTheme();
+  const { isValid, values } = useFormikContext();
+  const { newPassword, confirmPassword } = values as FormikValues; 
+  const [showPassword, setShowPassword] = useState(false);
+  const [showPasswordIcon, setShowPasswordIcon] = useState(false);
+
+  useAuthenticatedRedirect();
+
+  useEffect(()=> {
+    newPassword || confirmPassword ? setShowPasswordIcon(true) : setShowPasswordIcon(false);
+  },[newPassword, confirmPassword]);
+
+  const togglePassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const PasswordEndAdornment = () => {
+    const PasswordIcon = showPassword ? VisibilityOffIcon : VisibilityIcon;
+
+    return showPasswordIcon ? (
+      <IconButton
+        aria-label="Change password field visible status"
+        onClick={ togglePassword }
+      >
+        <PasswordIcon sx={ { color: theme.colors.white } } />
+      </IconButton>
+    ) : null;
+  };
```

Comment:
I noticed that it was neccessary to click the `PasswordEndAdornment` icon twice in order for it to register the click. This is because the component is being re-rendered on each render, so it isn't retaining it's state correctly. If you move the `PasswordEndAdornment` component outside of the `SignUp` function, it will resolve the issue of requiring multiple clicks.

I think putting the `PasswordEndAdornment` component in the `components` directory would make sense, since it will likely be required on the login screen as well.

Reply:
Good catch, i hadn't noticed that.

Maybe i can extract the whole password input and make that a component that has the toggle icon functionality in it too

Reply:
That sounds like a good approach to me. Might be tricky since the password and confirm password inputs have to be in sync though. May have to handle the toggle icon functionality externally